### PR TITLE
Feature/KITOSUDV-3615

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Presentation.Web/app/components/org/structure/org-unit-migration-table.component.ts [hunk 1]
```diff
@@ -108,6 +115,8 @@
 
             this.organizationUnitService.deleteSelectedRegistrations(this.organizationUuid, this.unitUuid, request)
                 .then(() => {
+                    this.stateParameters.registrationsChanged();
+
                     this.options.refreshData();
```
Comment (mrjsawdk) [inline Presentation.Web/app/components/org/structure/org-unit-migration-table.component.ts:120]: we have to talk about this line of code in a call.. it's hiding a promise making your state management rubbish :'(

File: Presentation.Web/app/components/org/structure/org-unit-migration.component.ts [hunk 2]
```diff
@@ -114,6 +114,8 @@
             const request = this.createChangeRequest();
             this.organizationUnitService.deleteSelectedRegistrations(this.organizationUuid, this.unitUuid, request)
                 .then(() => {
+                    this.stateParameters.registrationsChanged();
+
                     this.refreshData();
```
Comment (mrjsawdk) [inline Presentation.Web/app/components/org/structure/org-unit-migration.component.ts:119]: also an issue in this line

File: Presentation.Web/app/components/org/structure/org-unit-migration.component.ts [hunk 3]
```diff
@@ -140,6 +142,8 @@
             this.organizationUnitService.transferSelectedRegistrations(this.organizationUuid, this.unitUuid, request)
                 .then(() => {
                     this.selectedOrg = null;
+                    this.stateParameters.registrationsChanged();
+
                     this.refreshData();
```
Comment (mrjsawdk) [inline Presentation.Web/app/components/org/structure/org-unit-migration.component.ts:147]: and this line

**Review by mrjsawdk — COMMENTED** [review]
@a-naskret Let's talk about the comment in a call.. I will just test the code and then I will give you a "ring"

File: Presentation.Web/app/components/org/structure/org-unit-migration-table.component.ts [hunk 4]
```diff
@@ -71,6 +74,10 @@
                 console.error(`missing migration table attribute: 'organizationId' for table with title: ${this.title}`);
                 return;
             }
+            if (this.stateParameters === null) {
+                console.error(`missing migration table attribute: 'stateParameters' for table with title: ${this.title}`);
```
Comment (a-naskret) [inline Presentation.Web/app/components/org/structure/org-unit-migration-table.component.ts:78]: text to danish

File: Presentation.Web/app/components/org/structure/org-unit-migration-table.component.ts [hunk 5]
```diff
@@ -108,16 +115,18 @@
...
-            this.options.setIsBusy(false);
+                    this.stateParameters.registrationsChanged();
+
+                    return this.options.refreshData();
+                })
+                .then(
+                    () => this.options.setIsBusy(false),
+                    error => {
+                        console.error(error);
+                        this.notify.addErrorMessage("Failed to deleted the selected unit");
```
Comment (a-naskret) [inline Presentation.Web/app/components/org/structure/org-unit-migration-table.component.ts:126]: text to danish

File: Presentation.Web/app/components/org/structure/org-unit-migration.component.ts [hunk 6]
```diff
@@ -114,14 +114,17 @@
...
-                    this.setIsBusy(false);
-                });
+                    this.stateParameters.registrationsChanged();
+                    return this.refreshData();
+                })
+                .then(
+                    () => this.setIsBusy(false),
+                    error => {
+                        console.error(error);
+                        this.notify.addErrorMessage("Delete selected failed");
```
Comment (a-naskret) [inline Presentation.Web/app/components/org/structure/org-unit-migration.component.ts:124]: text to danish

File: Presentation.Web/app/components/org/structure/org-unit-migration.component.ts [hunk 7]
```diff
@@ -140,13 +143,18 @@
...
-                });
+                    this.stateParameters.registrationsChanged();
+
+                    return this.refreshData();
+                })
+                .then(
+                    () => this.setIsBusy(false),
+                    error => {
+                        console.error(error);
+                        this.notify.addErrorMessage("Transfer failed");
```
Comment (a-naskret) [inline Presentation.Web/app/components/org/structure/org-unit-migration.component.ts:154]: text to danish

**Review by mrjsawdk — APPROVED** [review]

**Review by mrjsawdk — APPROVED** [review]

**Pull request merged**